lib/Sema/ConstraintSystem.cpp

```diff
@@ -2806,7 +2806,9 @@ ConstraintSystem::getTypeOfMemberReference(
     FunctionType::ExtInfo info;
 
     if (Context.LangOpts.hasFeature(Feature::InferSendableFromCaptures)) {
-      if (isPartialApplication(locator) && baseOpenedTy->isSendableType()) {
+      if (isPartialApplication(locator) &&
+          (resolvedBaseTy->is<AnyMetatypeType>() ||
+           baseOpenedTy->isSendableType())) {
```

Comment:
Why does `isSendableType()` return `false` for `AnyMetatypeType`? That seems like a problem

Reply:
It's not that, in this example `Array.init` has unbound generic parameters so if we were to use `isSendableType` on that it would crash.

Reply:
And `baseOpenedTy` is an instance type...